app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -22,5 +23,17 @@ class MapFragment : Fragment(R.layout.fragment_map) {
         binding.logoutBtn.setOnClickListener {
             findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
         }
+
+        binding.ClearBtn.setOnClickListener {
+            initPersistence()
+            findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
+        }
+    }
+
+    private fun initPersistence() {
+        val sharedPref = activity?.getPreferences(Context.MODE_PRIVATE)
```

Comment:
Editing preferences can also work as a builder, I suggest using it like so:
```
requireActivity.getPreferences(MODE_PRIVATE)  // 1
    .putBoolean(KEY_IS_USER_LOGGED_IN, false) // 2
    .apply() // 3
```

1. use requireActivity() to avoid null check
2. Use a public static constant for consistency throughout your code. This way you need to change the key value only in one place
3. use `apply` to perform the write async and not block the UI thread

Reply:
Also you can import `MODE_PRIVATE` individually